fairscale/nn/moe/top2gate.py

```diff
@@ -57,8 +68,8 @@ def top2gating(logits: torch.Tensor) -> Tuple[Tensor, Tensor, Tensor]:
     mask2 *= torch.lt(locations2, capacity)
 
     # Store the capacity location for each token
-    locations1_gs = torch.einsum("gse,gse->gs", locations1, mask1)
-    locations2_gs = torch.einsum("gse,gse->gs", locations2, mask2)
+    locations1_gs = torch.sum(locations1 * mask1, dim=2)
```

Comment:
The earlier einsum did not work on the GPU:

E       RuntimeError: baddbmm for CUDA tensors only supports floating-point types. Try converting the tensors with .float() at /private/home/
user/src/pytorch/aten/src/THC/generic/THCTensorMathBlas.cu:324